lib/brcobranca/remessa/cnab400/banco_nordeste.rb

```diff
@@ -0,0 +1,165 @@
+# -*- encoding: utf-8 -*-
+module Brcobranca
+  module Remessa
+    module Cnab400
+      class BancoNordeste < Brcobranca::Remessa::Cnab400::Base
+        # documento do cedente
+        attr_accessor :documento_cedente
+        # 1 - Emitido pelo banco
+        # 2 - Emitido pelo cliente
+        attr_accessor :emissao_boleto
+
+        validates_presence_of :agencia, :conta_corrente, message: 'não pode estar em branco.'
+        validates_presence_of :documento_cedente, :digito_conta, message: 'não pode estar em branco.'
+        validates_length_of :agencia, maximum: 4, message: 'deve ter 4 dígitos.'
+        validates_length_of :conta_corrente, maximum: 7, message: 'deve ter 7 dígitos.'
+        validates_length_of :documento_cedente, minimum: 11, maximum: 14, message: 'deve ter entre 11 e 14 dígitos.'
+        validates_length_of :carteira, maximum: 2, message: 'deve ter 2 dígitos.'
+        validates_length_of :digito_conta, maximum: 1, message: 'deve ter 1 dígito.'
+
+        # Nova instancia do Banco do Nordeste
+        def initialize(campos = {})
+          campos = {
+            aceite: 'N',
+            carteira: '21',
+            emissao_boleto: '2'
+          }.merge!(campos)
+
+          super(campos)
+        end
+
+        def agencia=(valor)
+          @agencia = valor.to_s.rjust(4, '0') if valor
+        end
+
+        def conta_corrente=(valor)
+          @conta_corrente = valor.to_s.rjust(7, '0') if valor
+        end
+
+        def carteira=(valor)
+          @carteira = valor.to_s.rjust(2, '0') if valor
+        end
+
+        def cod_banco
+          '004'
+        end
+
+        def nome_banco
+          'B.DO NORDESTE'.ljust(15, ' ')
+        end
+
+        # Informacoes da conta corrente do cedente
+        #
+        # @return [String]
+        #
+        def info_conta
+          # CAMPO            TAMANHO
+          # agencia          4
+          # zeros            2
+          # conta corrente   7
+          # digito da conta  1
+          # complemento      6
+          "#{agencia}00#{conta_corrente}#{digito_conta}#{''.rjust(6, ' ')}"
+        end
+
+        # Complemento do header
+        # (no caso do Banco do Nordeste, sao apenas espacos em branco)
+        #
+        # @return [String]
+        #
+        def complemento
+          ''.rjust(294, ' ')
+        end
+
+        # Codigo da carteira de acordo com a documentacao do Banco do Nordeste
+        # se a carteira nao forem as testadas ()
+        # retorna 'I' que é o codigo das carteiras restantes na documentacao
+        #
+        # @return [String]
+        #
+        def codigo_carteira
+          if emissao_boleto.to_s == "1"
+            return "1" if carteira.to_s == "21"
+            return "2" if carteira.to_s == "41"
+          elsif emissao_boleto.to_s == "2"
+            return "4" if carteira.to_s == "21"
+            return "5" if carteira.to_s == "41"
+          end
+
+          "I" if carteira.to_s == "51"
+        end
```

Comment:
Muito return assim me deixa meio :drunk:

Dava pra fazer um negócio tipo assim, se não for deixar mais confuso
```ruby
carteiras = {
  "0" => { "51" => "I" },
  "1" => { "21" => "1", "41" => "2" },
  "2" => { "21" => "4", "41" => "5" }
}

if %w(1 2).include?(emissao_boleto)
  carteiras[emissao_boleto][carteira]
else
  carteiras["0"][carteira]
end
```